docs/sphinx/install.rst

```diff
@@ -586,6 +586,93 @@ inside the container. Note that most Python packages that use GPU-acceleration,
 
     python3 -m pip install cupy-cuda11x
 
+.. _distributed-computing-with-mpi:
+
+Distributed Computing with MPI
+------------------------------------
+
+CUDA Quantum supports the Message Passing Interface (MPI) parallelism via a runtime configurable plugin interface.
+Specifically, while a builtin plugin is shipped with the CUDA Quantum Docker image, users can also activate 
+a custom CUDA Quantum MPI plugin targeting their local MPI installation without recompiling or reinstalling the stack.
+
+CUDA Quantum MPI API for C++ and Python will then be delegated 
+to the plugin implementation selected at runtime.  
+
+The following table summarizes mechanisms whereby a CUDA Quantum MPI plugin can be picked up.
+
+.. tab:: Builtin
+
+  **Use case**: build from source, Docker containers
+
+  When a local MPI installation is detected at build time, CUDA Quantum will build an MPI plugin targeting this implementation when build from source.
+  Similarly, the :ref:`CUDA Quantum Docker image <install-docker-image>` is shipped with a builtin MPI plugin based on its optimized OpenMPI installation inside the image. 
+  No action is required to use this builtin plugin. 
+
+  .. note::
+
+    Manually-activated MPI plugin takes precedence over builtin plugin. 
+    For instance, the user may choose to reinstall a different MPI library inside the Docker container and 
+    activate a new CUDA Quantum MPI plugin. The newly-activated plugin will always be used even though a builtin one 
+    was shipped with the Docker image. 
+
+.. tab:: Manual Activation
+
+  **Use case**: data centers with custom MPI installation, CUDA Quantum binary distributions without MPI support (including Python wheels).
```

Comment:
Curious to why we don't have MPI support in binary distributions and Python wheels

Reply:
'CUDA Quantum binary distributions' here refers to general distributions that may or may not have MPI bundled. Python wheel distribution is one example where we don't currently have MPI in the distributed binary. 
I think the main reason is that the default Python `pip` package manager doesn't support non-Python packages, like MPI, as a package dependency.

Reply:
Re Python wheels: wheels have a pretty strict size limit, and it is hence expected that wheels do take dependencies on other wheels rather than include the same content. If you look at the [mpi4py page on PyPI](https://pypi.org/project/mpi4py/) you find that the wheel itself only contains the bindings, but expects an existing MPI installation on the system - like I already mentioned regarding CUDA. It hence makes little sense for us to declare that wheel as a dependency. A better way to get a fully functional MPI installation usable from Python is to follow the instructions we give, which install a *CUDA-aware!* MPI implementation along with the python bindings using conda.